functions.py

```diff
@@ -0,0 +1,201 @@
+        employees to a temporary file and then deletes the old employees file. Afterwards, it saves it as the main
+        employees file (employees). employees_name = a string.
+        """
+        employees_csv_data = self.open_and_read_file(self.list_of_employees)
+        with open(EMPLOYEES_EDIT_FILE, 'w') as updated_csv:
+            writer = csv.writer(updated_csv)
+            if str(new_employee.employee_id) not in str(employees_csv_data.values):
+                print(EMPLOYEE_DOESNT_EXIST_IN_FILE_MSG)
+            else:
+                for row in employees_csv_data.values:
+                    if new_employee.employee_id != str(row[0]):
+                        writer.writerow(row)
+        os.remove(self.list_of_employees)
+        os.rename(EMPLOYEES_EDIT_FILE, self.list_of_employees)
+
+    def delete_employees_from_file(self):
+        """
+        This function gets from the user a file path that contains the data of the employees and removes the
+        employees from the employees file only if all the data of all employees is supplied. employees_to_delete = a
+        string.
+        """
+        try:
+            employees_to_delete_file_path = raw_input(REMOVE_EMPLOYEES_FROM_FILE_MSG)
+            data_to_erase = self.open_and_read_file(employees_to_delete_file_path).values.tolist()
+            data_to_keep = self.open_and_read_file(self.list_of_employees).values.tolist()
+            with open(EMPLOYEES_EDIT_FILE, 'w') as updated_csv:
+                writer = csv.writer(updated_csv)
+                for row_data_to_erase in data_to_erase:
+                    for row in data_to_keep:
+                        if row[0] != row_data_to_erase[0] :
+                            writer.writerow(row)
+
+            os.remove(self.list_of_employees)
+            os.rename(EMPLOYEES_EDIT_FILE, self.list_of_employees)
+        except FileNotFoundError:
+            print(FILE_DOES_NOT_EXIST_MSG)
+
+    def mark_attendance(self, new_employee):
+        """
+        The function gets an employee's id as the input and saves the date and time of him in the attendance log file.
+        employee_id = a 9 numbers integer.
+        monthly_report_url = a string.
+        """
+        attendance_data = list()
+        attendance_data.append(new_employee.employee_id)
+        name_to_save = None
+        attendance_time = datetime.datetime.now()
+        attendance_time = attendance_time.strftime('%Y/%m/%d %H:%M:%S')
+        attendance_time = str(attendance_time)
+        attendance_data.append(attendance_time)
+        with open(self.list_of_employees) as File:
+            reader = csv.reader(File, delimiter=',', quotechar=',',
+                                quoting=csv.QUOTE_MINIMAL)
+            for row in reader:
+                if row:
+                    if str(new_employee.employee_id) == str(row[0]):
+                        name_to_save = row[1]
```

Comment:
"""
 name_to_save = row[1]
 attendance_data.append(str(name_to_save))
"""

pls write in one line

Reply:
done